app/uk/gov/hmrc/emailverification/services/AuditService.scala

```diff
@@ -30,38 +31,193 @@ class AuditService @Inject() (
     auditConnector: AuditConnector
 )(implicit ec: ExecutionContext) extends BackendHeaderCarrierProvider {
 
-  def sendPinViaEmailEvent(emailAddress: String, passcode: String, serviceName: String, responseCode: Int)(implicit request: Request[_]) = {
+  def sendPasscodeViaEmailEvent(emailAddress: String, passcode: String, serviceName: String, responseCode: Int)(implicit request: Request[_]) = {
     val details = Map(
       "emailAddress" -> emailAddress,
-      "pinCode" -> passcode,
+      "passcode" -> passcode,
       "serviceName" -> serviceName,
       "bearerToken" -> hc.authorization.getOrElse(Authorization("-")).value,
       "responseCode" -> responseCode.toString
     )
 
-    sendEvent("SendEmailWithPin", details, "Email Verification - Send out PIN via Email")
+    sendEvent("SendEmailWithPasscode", details, "HMRC Gateway - Email Verification - Send out passcode via Email")
   }
 
-  def sendCheckEmailVerifiedEventSuccess(emailAddress: String, passcode: String, responseCode: Int)(implicit request: Request[_]) = {
+  def sendCheckEmailVerifiedEvent(emailAddress: String, failureReason: Option[String], responseCode: Int)(implicit request: Request[_]) = {
     val details = Map(
-      "pinCode" -> passcode,
-      "emailVerified" -> "true",
+      "emailAddress" -> emailAddress,
+      "emailVerified" -> failureReason.fold("true")(_ => "false"),
+      "responseCode" -> responseCode.toString,
+      "bearerToken" -> hc.authorization.getOrElse(Authorization("-")).value
+    ) ++ failureReason.fold[Map[String, String]](Map()) { reason =>
+        Map("verifyFailureReason" -> reason)
+      }
```

Comment:
if you like abusing the scala compiler
```suggestion
    ) ++ failureReason.map("verifyFailureReason" -> _).toMap
``` 
works

Reply:
👍